src/Calypso-SystemPlugins-SUnit-Browser/ClyJumpToTestMethodCommand.class.st

```diff
@@ -0,0 +1,103 @@
+"
+Description
+--------------------
+
+I am a command available in the method context menu of Calypso. When activated I'll jump on the test method corresponding to the method. 
+
+If the method does not exists I will create it in the corresponding class.
+
+Internal Representation and Key Implementation Points.
+--------------------
+
+    Instance Variables
+	selectedMethods:			<aCollection>	Methods selected by the user that should have their tests generated.
+	testMethodToBrowse:		<aMethod>			The method to browse at the end of the execution of the command.
+
+"
+Class {
+	#name : #ClyJumpToTestMethodCommand,
+	#superclass : #SycMethodCommand,
+	#traits : 'TClyGenerateTestClass',
+	#classTraits : 'TClyGenerateTestClass classTrait',
+	#instVars : [
+		'testMethodToBrowse',
+		'browser',
+		'systemEnvironment'
+	],
+	#category : #'Calypso-SystemPlugins-SUnit-Browser-Commands'
+}
+
+{ #category : #testing }
+ClyJumpToTestMethodCommand class >> canBeExecutedInContext: aBrowserContext [
+	^ aBrowserContext isInstanceSideMethodSelected and: [ aBrowserContext selectedMethods anySatisfy: [ :each | each isTestMethod not ] ]
+]
+
+{ #category : #activation }
+ClyJumpToTestMethodCommand class >> methodMenuActivation [
+	<classAnnotation>
+	^ CmdContextMenuActivation byItemOf: ClySUnitMethodMenuGroup order: 10 for: ClyMethod asCalypsoItemContext
+]
+
+{ #category : #activation }
+ClyJumpToTestMethodCommand class >> methodShortcutActivation [
+	<classAnnotation>
+	^ CmdShortcutActivation by: $h meta , $j meta for: ClyMethod asCalypsoItemContext
+]
+
+{ #category : #accessing }
+ClyJumpToTestMethodCommand >> defaultMenuItemName [
+	^ 'Jump to test method'
+]
+
+{ #category : #execution }
+ClyJumpToTestMethodCommand >> execute [
+	self methods ifEmpty: [ self inform: 'No method selected'. ^ self ].
+	
+	self methods do: [ :method | self generateTestMethodFor: method ].
+	
+	testMethodToBrowse ifNotNil: [ :testMethod | browser showMethod: testMethod ]
```

Comment:
browser here is not a context instance but a browser. So it should be  #selectMethod:

Reply:
Are you sure?

```Smalltalk
ClyMethodContextOfFullBrowser someInstance respondsTo: #selectMethod:. "false"
ClyMethodContextOfFullBrowser someInstance respondsTo: #showMethod:. "true"
```

Reply:
Oh, but I have a context here. Maybe I did not select what you wanted me to select in the prepare method.

Reply:
Yes, look at next comment